### ISSUE TITLE
Update simplesamlphp

### ISSUE THREAD

**Issue by nadavkav** [catalyst/moodle-auth_saml2#744]
There are new upstream updates for the simplesamlphp lib.
Please see if you can get a new version for /.extlib/simplesamlphp.

**Comment by danmarsden** [comment]
Thanks @nadavkav - we're slightly more motivated when upstream contains security issues or when upstream fixes something we need - otherwise we tend up update the lib in an ad-hoc manner - there is a readme file here if you want to give it a go and send a PR?
https://github.com/catalyst/moodle-auth_saml2/blob/MOODLE_39_STABLE/.extlib/UPGRADE.md

**Comment by nadavkav** [comment]
Hi @danmarsden , thank you, I will upgrade locally and send a PR.

**Comment by catalyst-eu** [comment]
I suspect that Exceptions may not be thrown as anticipated based on this report from the PHP 8.0 PHPCS compliance checker. 
moodle-auth_saml2/.extlib/simplesamlphp/vendor/simplesamlphp/saml2/src/SAML2/Exception/RuntimeException.php
Line 12: The interface Throwable cannot be implemented directly, extend the Exception class instead.

(This is not a PHP8 thing)

**Comment by mark-webster-catalyst** [comment]
> Thanks @nadavkav - we're slightly more motivated when upstream contains security issues or when upstream fixes something we need - otherwise we tend up update the lib in an ad-hoc manner - there is a readme file here if you want to give it a go and send a PR?

Hi @danmarsden, We received this via our website contact form. I think they have a point. Would be good to find some time to review.

Hi! I am a developer at SAS Institute, and we are using your auth_saml2 plugin in a new Totara environment. Our IT Security team has run a Snyk scan on the saml2 code, and they have found a few high-level security vulnerabilities in the use of the simplesamlphp external library. I am working on following your instructions for updating the simplesamlphp library, however, I wanted to ensure that you were aware of these issues that exist with the version of that library in your current version of the SAML2 plugin.
1. Server-Side Request Forgery (SSRF) - Unsanitized input from an uploaded file flows into file_get_contents, where it is used as an URL to perform a request. This may result in a Server-Side Request Forgery vulnerability.
/.extlib/simplesamlphp/modules/admin/lib/Controller/Federation.php:408
2. Server-Side Request Forgery (SSRF) - Unsanitized input from an HTTP parameter flows into file_get_contents, where it is used as an URL to perform a request. This may result in a Server-Side Request Forgery vulnerability.
/.extlib/simplesamlphp/vendor/symfony/dependency-injection/EnvVarProcessor.php:122
3. Path Traversal - Unsanitized input from an uploaded file flows into file_get_contents, where it is used as a path. This may result in a Path Traversal vulnerability and allow an attacker to read arbitrary files.
/.extlib/simplesamlphp/modules/admin/lib/Controller/Federation.php:408
4. Path Traversal - Unsanitized input from an uploaded file flows into file_get_contents, where it is used as a path. This may result in a Path Traversal vulnerability and allow an attacker to read arbitrary files.
/.extlib/simplesamlphp/vendor/symfony/dependency-injection/EnvVarProcessor.php:122
5. Cross-site Scripting (XSS) - Unsanitized input from the HTTP request body flows into Response, where it is used to render an HTML page returned to the user. This may result in a Cross-Site Scripting attack (XSS).
/.extlib/simplesamlphp/vendor/symfony/http-kernel/HttpClientKernel.php:58
6. Cross-site Scripting (XSS) - Unsanitized input from an HTTP parameter flows into the echo statement, where it is used to render an HTML page returned to the user. This may result in a Cross-Site Scripting attack (XSS).
/.extlib/simplesamlphp/modules/statistics/lib/StatisticsController.php:163
7. Deserialization of Untrusted Data - Unsanitized input from an HTTP parameter flows into unserialize, where it is used to deserialize an object. This may result in an Unsafe Deserialization vulnerability.
/.extlib/simplesamlphp/lib/SimpleSAML/Auth/State.php:295
8. File Inclusion - Unsanitized input from an HTTP parameter flows into _, where it is included dynamically. Allowing unvalidated user input to control files that are included dynamically in PHP can lead to malicious code execution.
/.extlib/simplesamlphp/vendor/symfony/dependency-injection/EnvVarProcessor.php:124

**Comment by danmarsden** [comment]
Some of that looks like auto scan info that might trigger the same response on latest simplesamlphp lib to me... have they run snyk on latest simplesamlphp on its own?

**Comment by nadavkav** [comment]
I agree with @danmarsden that some of the warning are not serious as they seem to be, as it is coming from a static vulnerabilities code scan (which I am also running automatically for every MR) that has its limitations.
BUT still, it is better to have an updated simplesamlphp version, as much as we can, which includes fixes for all the issues that are reported on their upstream repo:
https://github.com/simplesamlphp/simplesamlphp
Which is at the moment set to 2.0.5

**Comment by danmarsden** [comment]
yeah - definitely worth updating if someone has the time to do this - pull requests always welcome! :-)

**Comment by matthewhilton** [comment]
Cleaning up issues - this is long been and done
